src/test/javascript/spec/component/assessment-dashboard/exercise-assessment-dashboard.component.spec.ts

```diff
@@ -343,32 +345,32 @@ describe('ExerciseAssessmentDashboardComponent', () => {
     });
 
     it('should set exam and stats properties', () => {
-        expect(comp.exam).toBeUndefined();
+        expect(comp.exam).toBe(undefined);
 
         comp.loadAll();
 
-        expect(comp.exercise).toBeDefined();
+        expect(comp.exercise).not.toBe(null);
```

Comment:
Does this cover `undefined` too? If not, it may need to be tested too.

Reply:
I agree `undefined` is not tested here. I think `.not.toBe(undefined)` would be better here (null values should not be used according to the documentation). However an even better assertion would be to check a field in the `exercise `object (maybe even the whole object). This would make sure that the value is neither `null` and `undefined` and it would also verify at least one value.

Reply:
changed accordingly!